dvc/scm/git/backend/dulwich.py

```diff
@@ -281,9 +281,11 @@ def ignore_manager(self):
     def is_ignored(self, path: str) -> bool:
```

Comment:
We can loosen the type-hinting here to allow path-like objects. If you update the imports at the top of this file to include
```
if TYPE_CHECKING:
    from dvc.types import StrPath
```

You can update this function declaration to:
```
def is_ignored(self, path: StrPath) -> bool
```
to allow the use of path-like objects. (`relative_path` will still always end up being a `str` after the `relpath()` call)

You will also have to update the function/typing declaration for `is_ignore` in the other `dvc/scm/git/backend/*.py` files